index/scorch/snapshot_index.go

```diff
@@ -430,3 +461,25 @@ func (i *IndexSnapshot) DocumentVisitFieldTerms(id index.IndexInternalID,
 
 	return nil
 }
+
+func stringsToMap(strsArr []string) map[string]bool {
+	if strsArr == nil {
+		return nil
+	}
+	strs := map[string]bool{}
+	for _, str := range strsArr {
+		strs[str] = true
+	}
+	return strs
+}
+
+func stringsRemoveStrings(stringArr, removeArr []string) []string {
```

Comment:
i find these string functions ugly, i would combine them into a single function.  while stringsToMap seems common, its also pretty trvial and probably clear to just manually inline everywhere.